src/mqtt.c

```diff
@@ -608,6 +634,10 @@ static int mqtt_config_publisher(oconfig_item_t *ci) {
       ERROR("mqtt plugin: Unknown config option: %s", child->key);
   }
 
+  if (conf->pattern != NULL) {
+    regcomp(&conf->topic_pattern, conf->pattern, REG_EXTENDED);
```

Comment:
Please check the return value of `regcomp()`. Regular expressions are easy to get wrong, and we have to report that fact to the user.